resolve/vector/Vector.cpp

```diff
@@ -30,28 +41,53 @@ namespace ReSolve { namespace vector {
   {
   }
 
+  /** 
+   * @brief destructor.
+   * 
+   */
   Vector::~Vector()
   {
     if (owns_cpu_data_) delete [] h_data_;
     if (owns_gpu_data_) mem_.deleteOnDevice(d_data_);
   }
 
 
+  /** 
+   * @brief get the number of elements in a single vector.
+   * 
+   * @return _n_, number of elements in the vector.
+   */
   index_type Vector::getSize() const
   {
     return n_;
   }
 
+  /** 
+   * @brief get the current number of elements in a single vector (use only for vectors with changing sizes, allocate for maximum expected size).
+   * 
+   * @return _n_current_, number of elements currently in thr vector.
+   */
   index_type Vector::getCurrentSize()
   {
     return n_current_;
   }
 
+  /** 
+   * @brief get the total number of vectors in multivector.
+   * 
+   * @return _k_, number of vectors in multivector, or 1 if the vector is not a multivector.
+   */
   index_type Vector::getNumVectors()
   {
     return k_;
   }
 
+  /** 
+   * @brief set the vector  data variable (HOST or DEVICE) to the provided pointer. IMPORTANT: this function DOES NOT ALLOCATE any data, it only assigns the pointer. 
+   * 
+   * @param[in] data     - Pointer to data
+   * @param[in] memspace - Memory space (HOST or DEVICE)  
+   */
```

Comment:
Perhaps use Doxygen `@warning` keyword to have this text nicely highlighted?
```suggestion
   * @brief set the vector  data variable (HOST or DEVICE) to the provided pointer. 
   * 
   * @param[in] data     - Pointer to data
   * @param[in] memspace - Memory space (HOST or DEVICE)  
   *
   * @warning This function DOES NOT ALLOCATE any data, it only assigns the pointer. 
   */
```